Elevator.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include "LeakedObjectDetector.h"
+
+struct Elevator
+{
+    int highestFloor = 50;
+    int lowestFloor = -2;
+    double totalWeightCarried = 0.0;
+    bool elevatorArrived = false;
+    double maxWeight;
+    int currentFloor;
+    double lastUserWeight = 0;
+    std::string name;
+
+    Elevator(std::string name);
+    ~Elevator();
+
+    void goToFloor(int designatedFloor);
+    void onboardPassenger(double userWeight);
+    void offboardPassenger();
+    bool canTakePassengerWeight(double userWeight);
+    void updateWeightCarried(double userWeight, bool userExit);
+    bool arrived();
+    void printTotalWeightCarried();
+	
+    struct ElevatorUser
+    {
+        double weight;
```

Comment:
`Elevator::ElevatorUser` is redundantly named. 
this class should simply be called `User`
- [ ] revise